### PR TITLE
Switch from JAX to PyTorch

### PR DESCRIPTION
**Pull request by jeremykubica**
Switch SigmaG filtering from JAX to PyTorch. The main advantage is better usability and control of PyTorch. An additional advantage is the PyTorch implementation is faster (1.7ms for 10,000 rows instead of 2.6ms on Baldur).

### CODE REVIEW

File: src/kbmod/filters/sigma_g_filter.py [hunk 1]
```diff
@@ -183,16 +130,45 @@ def compute_clipped_sigma_g_matrix(self, lh):
...
             or has been filtered (False).
         """
-        # We need to
-        os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
-        os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+        # Use a GPU if one is available.
+        if torch.cuda.is_available():
+            device = torch.device("cuda")
+        else:
+            device = torch.device("cpu")
```
Comment (maxwest-uw) [inline src/kbmod/filters/sigma_g_filter.py:137]: I'm curious if there's a difference in speed between using `jax` and `pytorch` on cpu

File: src/kbmod/filters/sigma_g_filter.py [hunk 2]
```diff
@@ -183,16 +130,45 @@ def compute_clipped_sigma_g_matrix(self, lh):
...
-        jnp_lh = jnp.array(lh)
-        jpn_inds_valid = self.sigma_g_jax_fn(jnp_lh).block_until_ready()
+        # Check whether the values fall within the bounds.
+        index_valid = torch.isfinite(torch_lh) & (torch_lh < upper_bnd) & (torch_lh > lower_bnd)
 
-        # Make sure the inds_valid is on CPU as a numpy array.
-        inds_valid = np.array(jpn_inds_valid)
-        return inds_valid
+        # Return as a numpy array on the CPU.
+        return index_valid.cpu().numpy()
```
Comment (maxwest-uw) [inline src/kbmod/filters/sigma_g_filter.py:171]: I assume that `pytorch` will handle this gracefully when running on a machine without a gpu in the first place?

**Review by maxwest-uw — APPROVED** [review]

File: src/kbmod/filters/sigma_g_filter.py (hunk 2 above)
Comment (jeremykubica) [reply src/kbmod/filters/sigma_g_filter.py:171]: Yes. If the tensor is already on the cpu, then it is a no-op.

File: src/kbmod/filters/sigma_g_filter.py (hunk 1 above)
Comment (jeremykubica) [reply src/kbmod/filters/sigma_g_filter.py:137]: Yes. On my laptop the pytorch version takes about 7ms and the JAX version takes about 100ms.
